cache/contenthash/checksum.go

```diff
@@ -619,3 +624,11 @@ func poolsCopy(dst io.Writer, src io.Reader) (written int64, err error) {
 	pool32K.Put(buf)
 	return
 }
+
+func convertPathToKey(p []byte) []byte {
+	return bytes.Replace([]byte(p), []byte("/"), []byte{0}, -1)
```

Comment:
Does it work for non-ASCII filename?

Reply:
What cases do you have in mind? All multibyte utf8 characters should start with a high bit on each byte.

Reply:
Windows may use non-Unicode charsets.
If Go (or some Win32 stuff) internally translates them to utf-8, there is no problem, but I'm not sure :sweat_smile:

Reply:
If that is the case we should convert the paths before they are used. Not for this case but just so that we have a stable checksum.